### ISSUE TITLE
Upgrade to 9 and a uiser got folder is not available (no files)

### ISSUE THREAD

**Issue by rpelissi** [owncloud/core#23052]
### Steps to reproduce
1. Have lastest of the 8.x owncloud installed on centos 7 (updated), using ldap auth
2. User have files in the folders (some users looks like <user>.<name>)
3.Upgrade to 9 using manual method

### Expected behaviour
After the upgrade the files should be visible

### Actual behaviour
No files appear and I have a message on the top saying "Folder not available". If I connect using the admin account I can create file and see them.

### Server configuration
**Operating system**: : CentOS Linux release 7.2.1511 (Core)

**Web server:** : Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips mod_fcgid/2.3.9 PHP/5.5.33 configured

**Database:** madiadb

**PHP version:** 5.5.33

**ownCloud version:** (see ownCloud admin page) : 9

**Updated from an older ownCloud or fresh install:**
Updated

**Where did you install ownCloud from:**
lastest 8.x
**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder
```
sudo -u apache php occ app:list
Enabled:
  - activity: 2.2.1
  - comments: 0.2
  - dav: 0.1.5
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - user_ldap: 0.8.0
Disabled:
  - encryption
  - external
  - files_external
  - user_external

**The content of config/config.php:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder
sudo -u apache php occ config:list system
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/www\/html\/dev\/owncloud\/data",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "d_owncloud",
        "dbhost": "lagrange.digitalrat.lan",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "50881d67e9310",
        "ldapIgnoreNamingRules": false,
        "loglevel": "2",
        "maintenance": false,
        "theme": "",
        "trusted_domains": [
            "192.170.6.6",
            "cloud.digitalrat.org"
        ],
        "mail_smtpmode": "smtp",
        "mail_smtphost": "localhost",
        "mail_smtpport": "25",
        "mail_from_address": "romain.pelissier",
        "mail_domain": "gmail.com",
        "secret": "***REMOVED SENSITIVE VALUE***"
    }

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your ownCloud installation folder
sudo -u apache php occ ldap:show-config
+-------------------------------+--------------------------------------------+
| Configuration                 |                                            |
+-------------------------------+--------------------------------------------+
| hasMemberOfFilterSupport      | 0                                          |
| hasPagedResultSupport         |                                            |
| homeFolderNamingRule          | opt:username                               |
| lastJpegPhotoLookup           | 0                                          |
| ldapAgentName                 | cn=Administrator,dc=digitalrat,dc=lan      |
| ldapAgentPassword             | ***                                        |
| ldapAttributesForGroupSearch  |                                            |
| ldapAttributesForUserSearch   |                                            |
| ldapBackupHost                |                                            |
| ldapBackupPort                | 389                                        |
| ldapBase                      | dc=digitalrat,dc=lan                       |
| ldapBaseGroups                | ou=group,dc=digitalrat,dc=lan              |
| ldapBaseUsers                 | ou=people,dc=digitalrat,dc=lan             |
| ldapCacheTTL                  | 600                                        |
| ldapConfigurationActive       | 1                                          |
| ldapDynamicGroupMemberURL     |                                            |
| ldapEmailAttribute            |                                            |
| ldapExperiencedAdmin          | 0                                          |
| ldapExpertUUIDGroupAttr       | auto                                       |
| ldapExpertUUIDUserAttr        | auto                                       |
| ldapExpertUsernameAttr        |                                            |
| ldapGroupDisplayName          | cn                                         |
| ldapGroupFilter               | objectClass=posixGroup                     |
| ldapGroupFilterGroups         | interne                                    |
| ldapGroupFilterMode           | 0                                          |
| ldapGroupFilterObjectclass    | posixGroup                                 |
| ldapGroupMemberAssocAttr      | member                                     |
| ldapHost                      | pdrdb001                                   |
| ldapIgnoreNamingRules         |                                            |
| ldapLoginFilter               | (&(|(objectclass=posixAccount))(uid=%uid)) |
| ldapLoginFilterAttributes     | cn                                         |
| ldapLoginFilterEmail          | 0                                          |
| ldapLoginFilterMode           | 0                                          |
| ldapLoginFilterUsername       | 1                                          |
| ldapNestedGroups              | 0                                          |
| ldapOverrideMainServer        |                                            |
| ldapPagingSize                | 500                                        |
| ldapPort                      | 389                                        |
| ldapQuotaAttribute            |                                            |
| ldapQuotaDefault              |                                            |
| ldapTLS                       | 0                                          |
| ldapUserDisplayName           | cn                                         |
| ldapUserDisplayName2          |                                            |
| ldapUserFilter                | (|(objectclass=posixAccount))              |
| ldapUserFilterGroups          |                                            |
| ldapUserFilterMode            | 0                                          |
| ldapUserFilterObjectclass     | posixAccount                               |
| ldapUuidGroupAttribute        | auto                                       |
| ldapUuidUserAttribute         | auto                                       |
| turnOffCertCheck              | 0                                          |
| useMemberOfToDetectMembership | 1                                          |
+-------------------------------+--------------------------------------------+

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**
Firefox 44.02

**Operating system:**
Windows 7 64

#### Web server error log
```
Insert your webserver log here
```
No error log from the webserver for what I see

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```
{"reqId":"VuBqL9TatHLZKfqY1XClsQAAAAU","remoteAddr":"192.170.6.2","app":"PHP","message":"Argument 2 passed to OC\\Files\\Cache\\Shared_Cache::__construct() must implement interface OCP\\Files\\Storage\\IStorage, null given, called in \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php on line 574 and defined at \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/cache.php#63","level":3,"time":"2016-03-09T18:23:44+00:00"}
{"reqId":"VuBqL9TatHLZKfqY1XClsQAAAAU","remoteAddr":"192.170.6.2","app":"PHP","message":"Argument 3 passed to OC\\Files\\Cache\\Shared_Cache::__construct() must implement interface OCP\\Files\\Cache\\ICacheEntry, null given, called in \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php on line 574 and defined at \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/cache.php#63","level":3,"time":"2016-03-09T18:23:44+00:00"}
{"reqId":"VuBqL9TatHLZKfqY1XClsQAAAAU","remoteAddr":"192.170.6.2","app":"PHP","message":"Call to a member function getCache() on a non-object at \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/cache.php#67","level":3,"time":"2016-03-09T18:23:44+00:00"}
{"reqId":"VuBqL1LUvZ5TGvw2XtIAkAAAAAk","remoteAddr":"192.170.6.2","app":"PHP","message":"Argument 2 passed to OC\\Files\\Cache\\Shared_Cache::__construct() must implement interface OCP\\Files\\Storage\\IStorage, null given, called in \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php on line 574 and defined at \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/cache.php#63","level":3,"time":"2016-03-09T18:23:45+00:00"}
{"reqId":"VuBqL1LUvZ5TGvw2XtIAkAAAAAk","remoteAddr":"192.170.6.2","app":"PHP","message":"Argument 3 passed to OC\\Files\\Cache\\Shared_Cache::__construct() must implement interface OCP\\Files\\Cache\\ICacheEntry, null given, called in \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php on line 574 and defined at \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/cache.php#63","level":3,"time":"2016-03-09T18:23:45+00:00"}
{"reqId":"VuBqL1LUvZ5TGvw2XtIAkAAAAAk","remoteAddr":"192.170.6.2","app":"PHP","message":"Call to a member function getCache() on a non-object at \/www\/html\/dev\/owncloud\/apps\/files_sharing\/lib\/cache.php#67","level":3,"time":"2016-03-09T18:23:45+00:00"}

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by LukasReschke** [comment]
This seems like another bug in `sharedstorage.php`

cc @icewind1991 @pv

**Comment by olerichter** [comment]
Hey I got the same error message, it goes away if you diable the shareing app, but i resolved it manualy by removing orphant shares from the database:


Server: Debian Testing 4.3.0-1-amd64
PHP: 5.6.17
Apache 2.4.18
Updated from 8.2 to 9.0

Server version 9.0.0.19 Debian Repository from owncloud.org

I have enabled:
LDAP
Share
External Storage
...

No encryption

**How I resolved it:**

For me the problem was caused by orphant shares between useres (created by owncloud 5 or 6, so very old) where ether the user (shared with) deleted the share folder from his directory or the shared folder was not existant in the user's home (who shared) because of eg. a removed external storage. Or if there where unavailible out of an other reason.

**Log:**
```

{PHP} Argument 2 passed to OC\Files\Cache\Shared_Cache::__construct() must implement interface OCP\Files\Storage\IStorage, null given, called in /var/www/owncloud/apps/files_sharing/lib/sharedstorage.php on line 574 and defined at /var/www/owncloud/apps/files_sharing/lib/cache.php#63
{PHP} Argument 3 passed to OC\Files\Cache\Shared_Cache::__construct() must implement interface OCP\Files\Cache\ICacheEntry, null given, called in /var/www/owncloud/apps/files_sharing/lib/sharedstorage.php on line 574 and defined at /var/www/owncloud/apps/files_sharing/lib/cache.php#63
PHP Fatal error:  Call to a member function getCache() on null in /var/www/owncloud/apps/files_sharing/lib/cache.php on line 67
apache2[16842]: {PHP} Call to a member function getCache() on null at /var/www/owncloud/apps/files_sharing/lib/cache.php#67
```

_additionaly I had this error massege which also resolved by removing orpant shares form the share table._
```
{webdav} Exception: {"Message":"File with id \"34489\" has not been found.","Exception":"OCP\\Files\\NotFoundException","Code":0,"Trace":"#0 \/var\/www\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php(82): OC\\Files\\View->getPath(34489)\n#1 \/var\/www\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php(570): OC\\Files\\Storage\\Shared->init()\n#2 \/var\/www\/owncloud\/lib\/private\/files\/storage\/wrapper\/wrapper.php(381): OC\\Files\\Storage\\Shared->getCache('', Object(OCA\\Files_Trashbin\\Storage))\n#3 \/var\/www\/owncloud\/lib\/private\/files\/view.php(1629): OC\\Files\\Storage\\Wrapper\\Wrapper->getCache()\n#4 \/var\/www\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php(82): OC\\Files\\View->getPath(33282)\n#5 \/var\/www\/owncloud\/apps\/files_sharing\/lib\/sharedstorage.php(570): OC\\Files\\Storage\\Shared->init()\n#6 \/var\/www\/owncloud\/lib\/private\/files\/storage\/wrapper\/wrapper.php(381): OC\\Files\\Storage\\Shared->getCache('', Object(OCA\\Files_Trashbin\\Storage))\n#7 \/var\/www\/owncloud\/lib\/private\/files\/view.php(1302): OC\\Files\\Storage\\Wrapper\\Wrapper->getCache('')\n#8 \/var\/www\/owncloud\/apps\/dav\/lib\/connector\/sabre\/serverfactory.php(124): OC\\Files\\View->getFileInfo('')\n#9 [internal function]: OCA\\DAV\\Connector\\Sabre\\ServerFactory->OCA\\DAV\\Connector\\Sabre\\{closure}(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#10 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)\n#11 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(446): Sabre\\Event\\EventEmitter->emit('beforeMethod', Array)\n#12 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#13 \/var\/www\/owncloud\/apps\/dav\/appinfo\/v1\/webdav.php(55): Sabre\\DAV\\Server->exec()\n#14 \/var\/www\/owncloud\/remote.php(138): require_once('\/var\/www\/ownclo...')\n#15 {main}","File":"\/var\/www\/owncloud\/lib\/private\/files\/view.php","Line":1639,"User":"XXXX"}
```

**Comment by icewind1991** [comment]
https://github.com/owncloud/core/pull/23039 should prevent breakage with orphan shares

**Comment by conloos** [comment]
Hi,

i have the same problem.
How can i scan for orphan shares?

Clean up the orphan shares seems the right solution for me.

Thanks Con

**Comment by olerichter** [comment]
Hey in #22976 in the first post they proposed a short term fix, should also apply to this issue or try the  icewind1991 fix as mentioned above. To "scan" for not working shares is a little complicated because you need check the log constantly, edit the DB and have a little knowlage about the shares and the users, and never the less **never try manual editing of the DB it on a production system**! it is very likely to cause serios problems. I only did it because it was a internal instance, no prodution system and I tried to gain knowlage for a propper bug report.

**Comment by PVince81** [comment]
Orphan shares should usually be cleant up by cron. If they don't, then they're likely not 100% orphaned and something is still connected somewhere.

**Comment by PVince81** [comment]
@olerichter please do these:
- `select * from oc_share where file_source=34489`
- `select * from oc_filecache where fileid=34489`
- `select * from oc_storages where numeric_id=$storage`, replace $storage with the value of the `storage` column from the previous result.

**Comment by PVince81** [comment]
> How can i scan for orphan shares?

**Comment by PVince81** [comment]
@conloos To find orphaned shares:

- from deleted file id: `select * from oc_share s where not exists (select fileid from oc_filecache fc where fc.fileid = s.file_source);`
- from deleted owner: `select * from oc_share s where not exists (select 1 from oc_users where s.uid_owner = uid);` (not for LDAP)
- shared with deleted recipient: `select * from oc_share s where not exists (select 1 from oc_users where s.share_with = uid);` (not for LDAP)
- shared with deleted group: `select * from oc_share s where not exists (select 1 from oc_groups where s.share_with = gid);` (not for LDAP)

**Comment by olerichter** [comment]
Will post the result on the weekend wenn I have a little sparetime again.

**Comment by PVince81** [comment]
@rullzer can you see if you can simulate these situations https://github.com/owncloud/core/issues/23052#issuecomment-197825127 in 9.0.0 and see whether it breaks anything ? Goal is to try and find out in what way the shares were orphaned.

**Comment by rullzer** [comment]
I tested on master. I did the 'destructive' operations directly on the db. So no cleanups or whatever.

* from deleted file id:
  1. share folder from user1 to user2
  2. delete folder from the filecache
  3. login as user1
  4. login as user2
* from deleted owner
  1. share folder from user1 to user2
  2. delete user1 from oc_users
  3. login as user2
* shared with deleted recipient: 
  1. share folder from user1 to user2
  2. delete user2 from oc_user
  3. login as user1
* shared with deleted group:
  1. share folder from user1 to group1
  2. delete group1 from oc_groups
  3. login as user1

This all worked for me. As in interface was still responsive etc.

Only in the case of deleted recipient and deleted group the share still showed up in for folder. Which is actually to be expected.

**Comment by PVince81** [comment]
@rullzer thanks a lot. So this tells us that the share entries from the bug reporters do not match these criteria.

Another possibility is that their shares are pointing at obsoleted storages, see what @blizzz found here: https://github.com/owncloud/core/issues/22976#issuecomment-197842198. Not sure if 100% related.

More information from @olerichter, @conloos and @rpelissi is required to confirm this.
Please run the following on your setups:
- `select * from oc_share where share_with='$user'`. Replace `$user` with the user id of group id of the user who is receiving the share and seeing the issue you reported.
- `select s.*, sh.*, fc.fileid, fc.path, fc.size from oc_share sh, oc_filecache fc, oc_storages s where sh.file_source=fc.fileid and fc.storage=s.numeric_id and share_with='$user';`
- `select * from oc_storages where id like '%$user%'`

It is likely that the "oc_storages" table contains two separate home entries for the user in question and the share is pointing at the wrong one.

**Comment by conloos** [comment]
@PVince81 

Not one or a group of users have the issue, all users dont see any file if sharing is enabled.

So, i see all sharings from the selected user on the first query.
No output on the second.
The last shows two entrys local:PATH_TO_USER and home::USER

If you need more informations i can give you access to my Server.

con

**Comment by PVince81** [comment]
> local:PATH_TO_USER

This could be bad. Does the path include your current "datadirectory" path (the one in config.php) or it is an older path ?

**Comment by conloos** [comment]
Hi,

no, you are right. I was blind. In the DB is my old layout.

'datadirectory' => '/var/www/owncloud/data',

select * from oc_storages where id like '%user%';
+-------------------------------+------------+-----------+--------------+
| id                            | numeric_id | available | last_checked |
+-------------------------------+------------+-----------+--------------+
| local::/var/www/own-data/user/ |          3 |         1 |         NULL |
| home::user                     |         34 |         1 |         NULL |
+-------------------------------+------------+-----------+--------------+

The Key "local" is wrong. Can i fix that direct in my DB or will i damage them?

Thanks Con

**Comment by PVince81** [comment]
@conloos now it is likely that maybe something is still linked to that one, possibly the broken shares.

Try these:
- `select * from oc_filecache where storage=3;`
- `select * from oc_share where file_source in (select fileid from oc_filecache where storage=3);`

**Comment by conloos** [comment]
MariaDB [owncloud]> select * from oc_filecache where storage=3;

+--------+---------+----------------------------------------------+----------------------------------+--------+------------------------------+----------+----------+---------+------------+-----------+---------------+------------------+---------------+-------------+----------+
| fileid | storage | path                                         | path_hash                        | parent | name                         | mimetype | mimepart | size    | mtime      | encrypted | etag          | unencrypted_size | storage_mtime | permissions | checksum |
+--------+---------+----------------------------------------------+----------------------------------+--------+------------------------------+----------+----------+---------+------------+-----------+---------------+------------------+---------------+-------------+----------+
| 121969 |       3 | files/Bilder/IMG_20141207_183239.jpg         | 023986aadc912f5e9b55db6dde6dd758 | 121962 | IMG_20141207_183239.jpg      |       14 |       12 | 1727601 | 1417974022 |         0 | 548491069d892 |                0 |    1417974022 |          27 | NULL     |
|  68782 |       3 | files/Dokumente/Arduino-cheat-sheet-v02c.pdf | 0aaa8455d8ddf838219122a33f6479c2 |  68781 | Arduino-cheat-sheet-v02c.pdf |        6 |        4 |  475590 | 1399575541 |         0 | 536bd3f571011 |                0 |    1399575541 |           0 | NULL     |
| 121963 |       3 | cache                                        | 0fea6a13c52b4d4725368f24b045ca84 |  59539 | cache                        |        1 |        2 |       0 | 1417974039 |         0 | 5484911704271 |                0 |    1417974039 |          31 | NULL     |
| 121975 |       3 | files/Bilder/IMG_20141207_183254.jpg         | 1be575649129a9dae0b0fb75a542bd9a | 121962 | IMG_20141207_183254.jpg      |       14 |       12 | 1320272 | 1417974039 |         0 | 548491171ae5d |                0 |    1417974039 |          27 | NULL     |
|  59541 |       3 | files_trashbin/files                         | 3014a771cbe30761f2e9ff3272110dbf |  59540 | files                        |        1 |        2 |       0 | 1399742695 |         0 | 5372392d53c9a |                0 |    1399742695 |           0 | NULL     |
| 121962 |       3 | files/Bilder                                 | 349e9c2afa1127f9d55a7492a3265ec5 |  14313 | Bilder                       |        1 |        2 | 6469436 | 1417974039 |         0 | 548491171d1bb |                0 |    1417974039 |          31 | NULL     |
|  14313 |       3 | files                                        | 45b963397aa40d4a0063e0d85e4fe7a1 |     -1 | files                        |        1 |        2 | 6945127 | 1417974039 |         0 | 548491171cdec |                0 |    1417973939 |          31 | NULL     |
|  59544 |       3 | files_trashbin/keyfiles                      | 728c7d8454e585e21166e28b78be8546 |  59540 | keyfiles                     |        1 |        2 |       0 | 1386801207 |         0 | 530f4deecbce5 |                0 |    1386801207 |           0 | NULL     |
|  68780 |       3 | files/readme.txt                             | 7f887bc5005abcb84b2e444a81f3e748 |  14313 | readme.txt                   |        7 |        8 |     101 | 1399574217 |         0 | 536bcec9a1cf4 |                0 |    1399574217 |          27 | NULL     |
|  93307 |       3 | files/Shared                                 | 82e4313655830e6a16e445f6665db47f |  14313 | Shared                       |        1 |        2 |      -1 | 1406376437 |         0 | 547485523da9b |                0 |    1406376437 |          31 | NULL     |
|  59543 |       3 | files_trashbin/versions                      | c639d144d3f1014051e14a98beac5705 |  59540 | versions                     |        1 |        2 |       0 | 1386801207 |         0 | 530f4deecb7f8 |                0 |    1386801207 |           0 | NULL     |
| 121972 |       3 | files/Bilder/IMG_20141207_183248.jpg         | cdc9ec912038abf0ca42375580df85b6 | 121962 | IMG_20141207_183248.jpg      |       14 |       12 | 1836638 | 1417974032 |         0 | 5484911060702 |                0 |    1417974032 |          27 | NULL     |
| 121966 |       3 | files/Bilder/IMG_20141207_183231.jpg         | d221b703f1602fa1cb840aaa55700dd1 | 121962 | IMG_20141207_183231.jpg      |       14 |       12 | 1584925 | 1417973977 |         0 | 548490d96780d |                0 |    1417973977 |          27 | NULL     |
|  59539 |       3 |                                              | d41d8cd98f00b204e9800998ecf8427e |     -1 |                              |        1 |        2 |       0 | 1399575541 |         0 | 536bd3f578bbf |                0 |    1386801207 |           0 | NULL     |
|  68781 |       3 | files/Dokumente                              | deb8a7a08780fc1a9c66dd9bf8ff4745 |  14313 | Dokumente                    |        1 |        2 |  475590 | 1399575541 |         0 | 536bd3f57844c |                0 |    1399575541 |          31 | NULL     |
|  59545 |       3 | files_trashbin/share-keys                    | efadb182f5229706b54ec20d20e8c46d |  59540 | share-keys                   |        1 |        2 |       0 | 1386801207 |         0 | 530f4deecc0fd |                0 |    1386801207 |           0 | NULL     |
|  59540 |       3 | files_trashbin                               | fb66dca5f27af6f15c1d1d81e6f8d28b |  59539 | files_trashbin               |        1 |        2 |       0 | 1386801207 |         0 | 530f4decb4679 |                0 |    1386801207 |          31 | NULL     |
+--------+---------+----------------------------------------------+----------------------------------+--------+------------------------------+----------+----------+---------+------------+-----------+---------------+------------------+---------------+-------------+----------+

MariaDB [owncloud]> select * from oc_share where file_source in (select fileid from oc_filecache where storage=3);
Empty set (0.01 sec)

**Comment by PVince81** [comment]
@conloos, ok then you should be able to delete this specific storage.
It looks like it is NOT related to the issue you're having.

**Comment by conloos** [comment]
sorry,

delete or replace the entrys?
It looks like if the id  'local' isnt in use anymore?

Thanks Con

**Comment by conloos** [comment]
Hi,

Sorry for the question, this was unnecessary.
I deleted the entrys and my problem is fixed. I see now all shares again.

Thanks Con

**Comment by PVince81** [comment]
> I see now all shares again.

Hmmmm... but that doesn't make sense. The shares weren't even pointing to any files there.

**Comment by PVince81** [comment]
> Will post the result on the weekend wenn I have a little sparetime again.

@olerichter did you get a chance ?

**Comment by PVince81** [comment]
@olerichter did you try with the patch from https://github.com/owncloud/core/pull/23039 ?

**Comment by icewind1991** [comment]
Afaik this is a duplicate of https://github.com/owncloud/core/issues/22976 and can be closed as it is fixed with #23039

**Comment by simsasaile** [comment]
Hi, I had this problem too, and the path #23039 did not help. I had to remove the invalid federated share in the mysql-database in table `oc_share_external`. Now I can log in again to my account. Before all I got was a blank page.
